cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsLaunchService.java

```diff
@@ -163,9 +176,35 @@
 
         awsTaggingService.tagRootVolumes(ac, amazonEC2Client, instances, stack.getTags());
 
+        if (stack.isCreateCloudWatchAlarm()) {
+            addCloudWatchAlarmsForSystemFailures(instances, regionName, credentialView);
+        }
         return awsResourceConnector.check(ac, instances);
     }
 
+    private void addCloudWatchAlarmsForSystemFailures(List<CloudResource> instances, String regionName, AwsCredentialView credentialView) {
```

Comment:
could you move cloudwatch related code/methods to a separate component?

Reply:
moved